Allow agents to be used for any tenant

kaikreuzer opened the pull request:
- specifying supported tenants for an agent is now optional
- removed oneapp as a channel from the tests

RobWin commented:
Thank you.
I think a test with an Agent which has not tenants would be helpful so that the new code is tested, or?

kaikreuzer commented:
The agents actually already had removed tenant sections.
I've now changed it to the billing agent referring to a tenant while the contract agent does not.
In the existing test, the channel has the same tenant as the billing agent and it successfully resolves both agents.
I've added an additional test with a channel that has a different tenant, which must not resolve the billing agent.
